src/compute/sort/primitive/indices.rs

```diff
@@ -18,16 +18,14 @@ where
     T: NativeType,
     F: Fn(&T, &T) -> std::cmp::Ordering,
 {
-    unsafe {
-        common::indices_sorted_unstable_by(
-            array.validity(),
-            |x: usize| *array.values().as_slice().get_unchecked(x),
```

Comment:
if the `as_slice` is the slow path, wouldn't it make sense to move it to outside the function here instead?

it seems to me that the current design requires allocating 2x the memory for dictionaries, strings, etc.

Reply:
Got it.